bundles/org.openhab.core.model.script.runtime/src/org/openhab/core/model/script/runtime/internal/engine/DSLScriptEngine.java

```diff
@@ -87,10 +87,10 @@ public Object eval(String script) throws ScriptException {
             if (script.stripLeading().startsWith(DSLScriptContextProvider.CONTEXT_IDENTIFIER)) {
                 String contextString = script.stripLeading().substring(
                         DSLScriptContextProvider.CONTEXT_IDENTIFIER.length(), script.stripLeading().indexOf('\n'));
-                String[] segments = contextString.split("-");
-                if (segments.length == 2) {
-                    modelName = segments[0];
-                    String ruleIndex = segments[1];
+                if (contextString.contains("-")) {
+                    int indexLastDash = contextString.lastIndexOf('-');
```

Comment:
Thanks. But I could be improved a little bit more. It is possible to skip the `contains("-")` check when calculation the `indexLastDash = lastIndexOf("-")` first and check for `-1` on it.

Reply:
I preferred it this way as it is imho better readable and the few cpu cycles should not really matter :-)

Reply:
Okay. Than I will agree on it too.